rocky/components/modal/script.js

```diff
@@ -0,0 +1,30 @@
+import { onDomReady } from "../js/imports/utils.js";
+
+onDomReady(initDialogs);
+
+export function initDialogs() {
+  let modal_components = document.querySelectorAll(".modal-wrapper");
+
+  modal_components.forEach((modal) => {
+    modal
+      .querySelector("button.modal-trigger")
+      .addEventListener("click", (event) => {
+        let target =
+          "dialog#" +
+          modal.querySelector("button.modal-trigger").dataset.target;
+        modal.querySelector(target).showModal();
```

Comment:
This does mean we need to introduce a new ID every time we add an extra modal on a page. That's a tricky little observation people might not make. A nearest Query would allow us to select the dialog next to the button in the same parent, while also removing the need for the dataset.target.

```suggestion
        event.target.closest('dialog').showModal();
```